src/pineko/theory.py

```diff
@@ -224,22 +225,62 @@ def opcard(self, name, grid, tcard):
         tcard : dict
             theory card
         """
-        opcard_path = self.operator_cards_path / f"{name}.yaml"
-        if opcard_path.exists():
-            if not self.overwrite:
-                rich.print(f"Skipping existing operator card {opcard_path}")
-                return
-        _x_grid, q2_grid = evolve.write_operator_card_from_file(
-            grid,
-            self.operator_cards_path
-            / configs.configs["paths"]["operator_card_template_name"],
-            opcard_path,
-            tcard,
-        )
-        if opcard_path.exists():
-            rich.print(
-                f"[green]Success:[/] Wrote card with {len(q2_grid)} Q2 points to {opcard_path}"
+        # First check if the fktable is for a DIS observable in an FONLL scheme
+        # because in this case multiple opcards need to be produced for the
+        # different components of the FONLL prescription.
+        # TODO: make sure this correctly produces the components we want (what about nf0 and nfref)
+        # TODO: extend for observables where we need 5 operator cards due to bottom (heracomb_b)
+        # TODO: correct pto for FONLL-B
+        pineapplgrid = pineappl.grid.Grid.read(grid)
+        if (
+            "yadism_version" in pineapplgrid.raw.key_values()
+            and "FONLL" in tcard["FNS"]
+        ):
+            opcard_paths = [
+                Path(str(self.operator_cards_path) + i) / f"{name}.yaml" for i in "123"
+            ]
+            theoryid = str(self.theory_id)
+            if tcard["FNS"] == "FONLL-B":
+                tcards = [
+                    {**tcard, "ID": theoryid + "1", "FNS": "FFNS", "NfFF": 3},
+                    {**tcard, "ID": theoryid + "2", "FNS": "FFN0", "NfFF": 3},
+                    {**tcard, "ID": theoryid + "3", "FNS": "FFNS", "NfFF": 4},
+                ]
+            else:
+                tcards = [
+                    {**tcard, "ID": theoryid + "1", "FNS": "FFNS", "NfFF": 3},
+                    {**tcard, "ID": theoryid + "2", "FNS": "FFN0", "NfFF": 3},
+                    {**tcard, "ID": theoryid + "3", "FNS": "FFNS", "NfFF": 4},
+                ]
```

Comment:
At the moment, these two branches seem defining the exact same value.

Reply:
They do, they're just FONLL with the PTO of the input theory as placeholder for the different FONLL schemes that I did not yet want to think about. In particular I still need to look at how I define the difference between the pto for evolution and the coefficients since that's relevant for FONLL-B. This distinction is no longer needed in yadism but I need a call to eko with a different pto for evolution than was used for the calculation is yadism, which is something I still need to figure out how (haven't really looked at pinefarm/eko yet). This is also what the last `TODO` in above is referencing.

Reply:
I would suggest to start directly producing an EKO theory for it, that will require moving the current translator from EKO in here.
https://github.com/NNPDF/eko/blob/3db8876f26de5c22bd59a90163f47eb0b232955a/src/eko/io/runcards.py#L168-L209

At that point, you will really have two different theories for EKO and pinefarm.

Reply:
Thanks, that would make sense.